substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/StackTraceUtils.java

```diff
@@ -123,7 +124,8 @@ public static boolean shouldShowFrame(FrameInfoQueryResult frameInfo, boolean sh
 
         if (!showReflectFrames && ((clazz == java.lang.reflect.Method.class && "invoke".equals(frameInfo.getSourceMethodName())) ||
                         (clazz == java.lang.reflect.Constructor.class && "newInstance".equals(frameInfo.getSourceMethodName())) ||
-                        (clazz == java.lang.Class.class && "newInstance".equals(frameInfo.getSourceMethodName())))) {
+                        (clazz == java.lang.Class.class && "newInstance".equals(frameInfo.getSourceMethodName())) ||
+                        (clazz == ClassInitializationInfo.class))) {
```

Comment:
No, I don't think we can do it that way. What are the failing cases here? What does `ClassInitializationInfo` have to do with reflection?

What you probably want is to annotate `ClassInitializationInfo` with `InternalVMMethod`, so that our internal class initialization logic is ignored in most stack walk. But that needs careful testing.

Reply:
Yes, that is what I'm trying to achieve, to ignore `ClassInitializationInfo` during stack walk. I'll try with `InternalVMMethod` annotation.